src/providers/hana/qgshanaprovider.cpp

```diff
@@ -1336,19 +1337,36 @@ QgsRectangle QgsHanaProvider::estimateExtent() const
   if ( conn.isNull() )
     return QgsRectangle();
 
+  if ( QgsHanaUtils::toHANAVersion(conn->getDatabaseVersion()).majorVersion() < 2 )
```

Comment:
This is incorrect. We need to check whether it is a HANA Cloud QRC1/2024 or HANA On-Premise SP8.

Reply:
I added a "proper" version check, which queries the cloud version from SYS.M_DATABASE.